PurchasesCoreSwift/Purchasing/OfferingsFactory.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  OfferingsFactory.swift
+//  PurchasesCoreSwift
+//
+//  Created by César de la Vega on 7/13/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+import StoreKit
+
+// TODO (Post-migration): Remove @objc and make it internal again.
+@objc(RCOfferingsFactory) public class OfferingsFactory: NSObject {
+
+    @objc public func createOfferings(withProducts products: [String: SKProduct],
+                                      data: [String: Any]) -> Offerings? {
+        guard let offeringsData = data["offerings"] as? [[String: Any]] else {
+            return nil
+        }
+
+        let offerings = offeringsData.reduce([String: Offering]()) { (dict, offeringData) -> [String: Offering] in
+            var dict = dict
+            if let offering = createOffering(withProducts: products, offeringData: offeringData) {
+                dict[offering.identifier] = offering
+            }
+            return dict
```

Comment:
this could get even fancier by first mapping it to `createOffering`, then reducing with the key